mim/commands/install.py

```diff
@@ -1,580 +1,303 @@
+    'Deprecated, will have no effect.')
 def cli(
-    package: str,
-    find_url: str = '',
-    timeout: int = 30,
+    args: Tuple[str],
+    index_url: Optional[str] = None,
     is_yes: bool = False,
-    is_user_dir: bool = False,
-    is_editable: bool = False,
 ) -> None:
-    """Install package.
+    """Install packages.
 
-    Example:
+    You can use `mim install` in the same way you use `pip install`!
 
-    \b
-    # install latest version of mmcv-full
-    > mim install mmcv-full  # wheel
-    # install 1.3.1
-    > mim install mmcv-full==1.3.1
-    # install master branch
-    > mim install mmcv-full -f https://github.com/open-mmlab/mmcv.git
-
-    # install latest version of mmcls
-    > mim install mmcls
-    # install 0.11.0
-    > mim install mmcls==0.11.0  # v0.11.0
-    # install master branch
-    > mim install mmcls -f https://github.com/open-mmlab/mmclassification.git
-    # install local repo
-    > git clone https://github.com/open-mmlab/mmclassification.git
-    > cd mmclassification
-    > mim install .
-
-    # install extension based on OpenMMLab
-    > mim install mmcls-project -f https://github.com/xxx/mmcls-project.git
-    """
-    install(
-        package,
-        find_url,
-        timeout,
-        is_yes=is_yes,
-        is_user_dir=is_user_dir,
-        is_editable=is_editable)
-
-
-def install(package: str,
-            find_url: str = '',
-            timeout: int = 15,
-            is_yes: bool = False,
-            is_user_dir: bool = False,
-            is_editable: bool = False) -> None:
-    """Install a package by wheel or from github.
+    And `mim install` will **install the 'mim' extra requirements**
+    for OpenMMLab package if needed.
 
-    Args:
-        package (str): The name of installed package, such as mmcls.
-        find_url (str): Url for finding package. If finding is not provided,
-            program will infer the find_url as much as possible. Default: ''.
-        timeout (int): The socket timeout. Default: 15.
-        is_yes (bool): Don’t ask for confirmation of uninstall deletions.
-            Default: False.
-        is_usr_dir (bool): Install to the Python user install directory for
-            environment variables and user configuration. Default: False.
-        is_editable (bool): Install a package in editable mode. Default: False.
+    \b
+    Example:
+        > mim install mmdet mmcls
+        > mim install git+https://github.com/open-mmlab/mmdetection.git
+        > mim install -r requirements.txt
+        > mim install -e <path>
+        > mim install mmdet -i <url> -f <url>
+        > mim install mmdet --extra-index-url <url> --trusted-host <hostname>
+
+    Here we list several commonly used options.
+    For more options, please refer to `pip install --help`.
     """
-    target_pkg, target_version = split_package_version(package)
+    if is_yes:
+        echo_warning(
+            'The `--yes` option has been deprecated, will have no effect.')
```

Comment:
Why do we deprecate the option?

Reply:
`pip install` does not ask y/n question, so we don't need this, see https://github.com/pypa/pip/issues/9750

Reply:
ok